lang/cs/Org.Apache.REEF.Wake/Avro/MessageInstance.cs

```diff
@@ -18,19 +18,18 @@
 namespace Org.Apache.REEF.Wake.Avro
 {
     /// <summary>
-    /// Wrapper class to bind a specific instance of an Avro messagage
-    /// with it associated sequence number.
+    /// Wrapper class to bind a specific instance of a message with the associated sequence number.
     /// </summary>
-    /// <typeparam name="T"></typeparam>
-    public struct MessageInstance<T>
+    /// <typeparam name="T">Message payload</typeparam>
+    public class MessageInstance<T> : IMessageInstance<T>
```

Comment:
I am not sure it can be done - we actually inherit from that class in one place, and I think it should be public because we also use it as is in the new Avro bridge.